### DISCUSSION TITLE
UI Package Manager

### DISCUSSION THREAD

**Discussion by aguilaair** [leoafarias/sidekick#121] [Ideas]
Instead of having a packages page with generic info about the the packages, I suggest having this UI be project based, so that users can add, remove or update their packages directly through sidekick.

This could also be related to #88 where Sidekick can edit many aspects of its projects.

----

# Problem Space
UI Package Manager

## Problem Statement
What problem or need are you trying to solve or fulfill?
Having to resort to the CLI and/or having to edit the pubspec to update, add and remove packages.

## Impact of the problem
What's the impact of this problem on our users and to Sidekick project?
Most are accustomed to using CLI tools, but this would give a high-level overview of all the packages in the project and give more detail than the CLI would give in a single command. Additionally, it would be faster if you're juggling multiple projects.

## How do we judge success?
What are the goals of the project and the success criteria by which they will be measured? (These need to be specific and measurable before moving into make it)
Being able to manage packages for flutter projects

## Possible Solutions
List your high-level ideas for possible solutions. (Can be filled out later)

# Validation
## What we already know?
What data or insights do you have to validate this?

## What do we need to answer?
What assumptions are we making that need to be validated/refuted? What questions will increase our confidence in the decisions we need to make? What are the gaps in our understanding?

People want a UI manager for their packages.

# Ready to make it

## What are we doing?
What will the project deliver? For example, an elevator pitch or read me.

A complete package manager for flutter/dart projects.

## Why will a user want this?
It's good to get Sidekick user's feedback on this.

To simplify adding, removing and updating packages

## Visualize the solution
It's good to get Design input here: whiteboard sketches, journey map, end to end demo

TBD

## Scale and scope
It's good to get Engineering input here: t-shirt size estimate

We already have package details for every project, we would have to design a UI to show it and the backend to edit the pubspec and search pub.dev

**Comment by leoafarias** [comment]
@aguilaair This could be interesting. I suggest filling out the idea template here https://github.com/leoafarias/sidekick/discussions/73, so we can properly plan for it. I have created this package which I have been using, which we can leverage if it makes sense. https://pub.dev/packages/pkg

Once the template is created I think we will have better visibility on what is needed.

**Reply by aguilaair** [reply]
Created one, has to be filled with more details but it's a good starting point
